### PR TITLE
Adds Span::log for key-value logging.

### PR DESCRIPTION
**Pull request by jcchavezs**
This is the key-value logging for spans.

Ping @felixfbecker

### CODE REVIEW

File: src/OpenTracing/LogUtils.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+{
+    $fields = [];
+
+    foreach ($keyValueFields as $key => $value) {
+        if (!is_string($key)) {
+            throw new InvalidArgumentException(sprintf("Non-string key, got %s of type %s", $key, gettype($key)));
+        }
+
+        switch (true) {
+            case is_bool($value):
```
Comment (felixfbecker) [inline src/OpenTracing/LogUtils.php]: iirc `$value === (bool)$value` is faster (for scalar types)

File: src/OpenTracing/LogUtils.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            case is_float($value) || is_numeric($value):
+                $fields[] = LogField::asFloat($key, $value);
+                break;
+            case is_int($value):
+                $fields[] = LogField::asInt($key, $value);
+                break;
+            case is_string($value):
+                $fields[] = LogField::asString($key, $value);
+                break;
+            case ($value instanceof Exception):
```
Comment (felixfbecker) [inline src/OpenTracing/LogUtils.php]: `Exception` is also `instanceof Throwable`

Comment (jcchavezs) [reply src/OpenTracing/LogUtils.php]: Not in versions prior 7.0. So in this version I use a polifyll for `Throwable` but that is only the interface declaration.
In PHP 5.6 + polifyll, exception does not implement Throwable.

File: src/OpenTracing/LogUtils.php (hunk 1 above)
Comment (jcchavezs) [inline src/OpenTracing/LogUtils.php]: Nice one. I will change it.

File: src/OpenTracing/LogUtils.php (hunk 2 above)
Comment (felixfbecker) [reply src/OpenTracing/LogUtils.php]: Gotcha. Would use a switch fallthrough then as it's the same logic

Comment (jcchavezs) [reply src/OpenTracing/LogUtils.php]: That is right. Changing it.

File: src/OpenTracing/LogUtils.php (hunk 1 above)
Comment (jcchavezs) [reply src/OpenTracing/LogUtils.php]: @felixfbecker I am willing to change it but I struggled when looking for references about the performance here. Do you have anyone?

Comment (felixfbecker) [reply src/OpenTracing/LogUtils.php]: Can't find it anymore, read it a long time ago. I know that function calls have a bigger overhead in PHP than in any other language so anything trivial that avoids a function call is significantly faster if called often enough

Comment (jcchavezs) [reply src/OpenTracing/LogUtils.php]: I see. I just changed it. Great catch!

File: src/OpenTracing/LogUtils.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    foreach ($keyValueFields as $key => $value) {
+        if (!is_string($key)) {
+            throw new InvalidArgumentException(sprintf("Non-string key, got %s of type %s", $key, gettype($key)));
+        }
+
+        switch (true) {
+            case $value === (bool) $value:
+                $fields[] = LogField::asBool($key, $value);
+                break;
+            case $value === (float) $value || is_numeric($value):
```
Comment (felixfbecker) [inline src/OpenTracing/LogUtils.php:23]: wouldn't the `is_numeric` prevent the `int` case from ever be reached?

**Pull request merged**